codechef DOM fix

jarifibrahim opened the pull request:
There are two elements with `custom-login`  and only one with `user-bar` . So changed it to `user-bar`. Also the `user-bar` element has a link to `logout`. So used it to check whether the user is logged in.

jarifibrahim committed:
codechef DOM change

ashish1294 commented:
@jarifibrahim : CodeChef is changing their front-end quite often recently.

Thanks for this ! Can you add your name to contributor list on README ?

jarifibrahim committed:
update readme.md and fix directory paths

jarifibrahim committed:
Update README.md

jarifibrahim committed:
Add support for HackerEarth and bug fixes

The plugin didn't work on SPOJ due to incorrect DOM access so fixed it.
The port.onMessage.addListener(function(arg){}) method of back.js required a JSON object as its argument
so changed prop.py to send messages to back.js in the form of json objects.
Added better exception handling with multiple try and except blocks and report exceptions to back.js

jarifibrahim committed:
Merge pull request #1 from jarifibrahim/hackerearth

Add support for HackerEarth and bug fixes

ashish1294 commented:
@jarifibrahim This is a great addition to the repo ! Thanks for this.
